src/js/node/events.js

```diff
@@ -92,9 +92,11 @@ const emitWithoutRejectionCapture = function emit(type, ...args) {
   if (events === undefined) return false;
   var handlers = events[type];
   if (handlers === undefined) return false;
-
-  for (var handler of [...handlers]) {
-    handler.apply(this, args);
+  // iterate in reverse so the handlers array does not need to be cloned
+  for (let i = handlers.length - 1; i >= 0; --i) {
```

Comment:
☝️ Node has prependListener and addListener so cloning would need to take place as the listeners can be modified while iterating.

Reply:
Ah, you mean the case where the called callback may actually add new listeners to the event? You are right, this might mess up this logic. So back to array cloning it is. 😕 I can also add a new test for this since this PR passes all tests atm.
(At work atm, will check your other comments later 🙂)